# Generate shared library with intel compilers
# March 9th 2018 by Sam Moors (Vrije Universiteit Brussel - VUB)
--- Makefile.intel.orig	2018-01-22 09:33:18.000000000 +0100
+++ Makefile.intel	2018-03-09 11:03:43.919624331 +0100
@@ -7,7 +7,7 @@
 CXX=icpc
 CC=icc
 # Turn of messages that hide real errors
-CFLAGS= -Wall -g -O2 -wd981 -wd279 -wd383 -vec-report0 -wd1572 -wd177
-CXXFLAGS=$(CFLAGS) -fno-rtti -fno-exceptions
+CFLAGS := $(CFLAGS) -wd981 -wd279 -wd383 -wd1572 -wd177
+CXXFLAGS := $(CXXFLAGS) -wd981 -wd279 -wd383 -wd1572 -wd177 -fno-rtti -fno-exceptions
 
 include Makefile.common
--- Makefile.common.orig	2018-01-22 09:33:18.000000000 +0100
+++ Makefile.common	2018-03-09 11:13:42.688253823 +0100
@@ -3,16 +3,20 @@
 
 PATHS:=-Iinclude -Isrc -Isrc/taylor -Isrc/functionals 
 CFLAGS+=$(PATHS)
+CXXFLAGS+=$(PATHS)
 LIBS+=-Llib
 
 BASE_OBJ:=src/xcint.o src/xcfun.o src/fortran.o
 FUN_OBJ:= $(patsubst %.cpp,%.o,$(wildcard src/functionals/*.cpp))
 
-lib: lib/libxcfun.a
+lib: lib/libxcfun.a lib/libxcfun.so
 
 lib/libxcfun.a: $(BASE_OBJ) $(FUN_OBJ)
 	ar -r $@ $(FUN_OBJ) $(BASE_OBJ)
 
+lib/libxcfun.so: $(BASE_OBJ) $(FUN_OBJ)
+	$(CXX) $(CXXFLAGS) -shared -o $@ $(FUN_OBJ) $(BASE_OBJ)
+
 genfortran: src/genfortran.cpp lib/libxcfun.a
 	$(CXX) $(CXXFLAGS) $^ -o $@ -lxcfun -lm $(LIBS)
 
